tests/src/JIT/HardwareIntrinsics/Add.cs

```diff
@@ -0,0 +1,403 @@
+                }
+
+                if (!shortTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE2 Add failed on short:");
+                    foreach (var item in shortTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+
+                if (!ushortTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE2 Add failed on ushort:");
+                    foreach (var item in ushortTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+
+                if (!floatTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE Add failed on float:");
+                    foreach (var item in floatTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+
+                if (!doubleTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE2 Add failed on double:");
+                    foreach (var item in doubleTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+
+                if (!sbyteTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE2 Add failed on sbyte:");
+                    foreach (var item in sbyteTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+
+                if (!byteTable.CheckResult((x, y, z) => x + y == z))
+                {
+                    Console.WriteLine("SSE2 Add failed on byte:");
+                    foreach (var item in byteTable.outArray)
+                    {
+                        Console.Write(item + ", ");
+                    }
+                    Console.WriteLine();
+                    testResult = Fail;
+                }
+            }
+
+            return testResult;
+        }
+
+        public struct TestTable<T> where T : struct
+        {
+            public T[] inArray1;
+            public T[] inArray2;
+            public T[] outArray;
+            public TestTable(T[] a, T[] b, T[] c)
+            {
+                this.inArray1 = a;
+                this.inArray2 = b;
+                this.outArray = c;
+            }
```

Comment:
Pin the arrays so you can safely grab the pointers without risk of GC hole?

```csharp
public struct TestTable<T> : IDisposable where T : struct
{
    public T[] inArray1;
    public T[] inArray2;
    public T[] outArray;
	
    GCHandle inHandle1;
    GCHandle inHandle2;
    GCHandle outHandle;
    
    public TestTable(T[] a, T[] b, T[] c)
    {
        this.inArray1 = a;
        this.inArray2 = b;
        this.outArray = c;
		
        inHandle1 = GCHandle.Alloc(inArray1, GCHandleType.Pinned);
        inHandle2 = GCHandle.Alloc(inArray2, GCHandleType.Pinned);
        outHandle = GCHandle.Alloc(outArray, GCHandleType.Pinned);
    }
	
    void Dispose()
    {
        inHandle1.Free();
        inHandle2.Free();
        outHandle.Free();
    }
	
    // CheckResult ...
}
```

Reply:
Good point, thank you! Will change.

Reply:
Could also expose the pointers via the struct if you want to skip 
`Unsafe.AsPointer(ref floatTable.inArray1[0])`

e.g.
```csharp
public void* inArray1Ptr => inHandle1.AddrOfPinnedObject().ToPointer();
public void* inArray2Ptr => inHandle2.AddrOfPinnedObject().ToPointer();
public void* outArrayPtr => outHandle.AddrOfPinnedObject().ToPointer();
```

So rather than 
```
var vf1 = Unsafe.Read<Vector256<float>>(Unsafe.AsPointer(ref floatTable.inArray1[0]));
```
Could do
```
var vf1 = Unsafe.Read<Vector256<float>>(floatTable.inArray1Ptr);
```

Reply:
That would be more elegant. Thanks for the suggestion.